### PR TITLE
Load ckeditor.js only when needed in Layout.pm

### PR DESCRIPTION
**Pull request by AjaySDwivedi1**

### CODE REVIEW

File: Kernel/Output/HTML/Layout.pm [hunk 1]
```diff
@@ -93,6 +93,7 @@ sub new {
     # allocate new hash for object
     # %Param often has a entry for 'SetCookies'
     my $Self = bless {%Param}, $Type;
+    my $HasRichTextEditor = $Self->{'Frontend::RichText'};
```
Comment (bschmalhofer) [inline Kernel/Output/HTML/Layout.pm:96]: The sysconfig setting is the general information that rich text editor is activated. But we need the information specific for a frontend module.

File: Kernel/Output/HTML/Layout.pm [hunk 2]
```diff
@@ -374,14 +375,15 @@ EOF
...
+            Key   => 'RichText',
             Value => 0,
         );
     }
 
     # check if rich text is active
-    if ( !$ConfigObject->Get('Frontend::RichText') ) {
-        $Self->{BrowserRichText} = 0;
+    if (!$HasRichTextEditor) {
+        $Self-> Footer();
```
Comment (bschmalhofer) [inline Kernel/Output/HTML/Layout.pm:385]: I dont't understand why Footer() is called here.

File: Kernel/Output/HTML/Layout.pm [hunk 3]
```diff
@@ -1741,7 +1743,7 @@ sub Footer {
     my ( $Self, %Param ) = @_;
 
     my $Type          = $Param{Type}           || '';
-    my $HasDatepicker = $Self->{HasDatepicker} || 0;
+    my $picker = $Self->{HasDatepicker} || 0;
```
Comment (bschmalhofer) [inline Kernel/Output/HTML/Layout.pm:1746]: This is a lexical variable. The name `$picker` is perfectly fine in Perl. But the OTOBO coding conventions state that variables should be in camel case.

File: Kernel/Output/HTML/Layout.pm (hunk 1 above)
Comment (AjaySDwivedi1) [reply Kernel/Output/HTML/Layout.pm:96]: Would something like $HasRichTextEditor = $Self->{'Frontend::RichText::CKEditor'} be more specific?

Comment (bschmalhofer) [reply Kernel/Output/HTML/Layout.pm:96]: The specifity is not the issue. OTOBO has a pretty elaborate configuraion system. Take a look at https://github.com/RotherOSS/otobo/blob/f8e9f730af50ee821fbf88f3eff033b706c28604/Kernel/Config/Files/XML/Framework.xml#L376 . The setting Frontend::RichText simply indicates that support for a rich text editor is active. But for all practical purposes, RTE is always enabled.  So this information is not really interesting.

Perhaps it is easier when actually looking at the installation. You can go to https://otobo.de/en/open-source-ticketing-system/#demos and log in as 'Lena/Lena'. You can change the user language at top left. Then you can choose Tickets->New Phone Ticket  to get to a page that has an embedded editor. Looking at the URL you see "Action=AgentTicketPhone". This gives you the information that the frontend module _Kernel/Modules/AgentTicketPhone.pm_ needs to set the flag 'HasRichTextEditor'.

Comment (AjaySDwivedi1) [reply Kernel/Output/HTML/Layout.pm:96]: Would it be something like this my $HasRichTextEditor = $ConfigObject->Get("Frontend::$Self->{Action}")?

Comment (bschmalhofer) [reply Kernel/Output/HTML/Layout.pm:96]: No, trial and error is not productive here. 

Perhaps it is better to move away from the actual changes and get a consensus of what should be achieved. Could you go back to #3223 and describe in your own words what should be achieved and how you plan to go ahead with it? In this way we can achieve a common understanding. But please don't despair when you are overwhelmed. OTOBO has an history of over 20 years and things are sometimes a bit peculiar in OTOBO.

**Comment by bschmalhofer** [comment]
I think that a new approach is needed here.

**Pull request closed without merging**